app/operations/base/activity_recording.rb

```diff
@@ -1,12 +1,14 @@
 module Base
   class ActivityRecording
-    def self.create_activity(model, current_user)
-      Activity.create(
-        project: fetch_project(model),
-        user: current_user,
-        action: self.class.name.split('::').last.downcase,
-        subject: model
-      )
+    def self.create_activity(model, current_user, action)
+      Array(model).map do |record|
+        Activity.create(
+          project: fetch_project(record),
+          user: current_user,
+          action: action,
+          subject: record
+        )
+      end
     end
 
     def self.fetch_project(model)
```

Comment:
this method is used outside this class? If it isn't you can make it private. Another question is why you're using class method instead of instance methods?

Reply:
I guess in the previous implementation, classes were heavily using interfaces, so the method calls were something like:

```rb
nofity_users
nofity_member
```
using class methods if is way more explicit 
```rb
UserNotification.notify_users
MemberNotification.notify_members
```

Reply:
The `self.fetch_project` is set to private_class_method after its implementation
And I think that a instance method cannot be used inside a class method